shielder/README.md

```diff
@@ -0,0 +1,136 @@
+You will need to paste the address in the 'Contract Address' field. Once you do that, two more fields will show up. You can choose any name of your liking for the 'Contract Name'. For the metadata, however, you will need to upload the `public_token/target/ink/public_token.json` file.
+After completing all that and clicking 'Add contract', you will be able to interact with the PSP22 contract.
+
+To call the contract as either Damian or Hans, you will need to click the `Caller` field and, despite its lack of responsiveness, paste the appropriate address there. Let's use `5D34dL5prEUaGNQtPPZ3yN5Y6BnkfXunKXXz6fo7ZJbLwRRH` (Damian) for the sake of this tutorial.
+
+This will allow you to easily check your balance by selecting the `balanceOf` method and again pasting Damian's address. This way we can check the balance before and after the withdrawal from Shielder and be extra sure the the funds actually ended up in the right account.
+
+### Withdrawing assets
+
+Withdrawing assets is in every sense dual to depositing them. We will run a command and see the funds appear back in our balances of the corresponding PSP22 tokens.
+
+Without further ado, let's run the following command (as Damian, so typing `//0` as the seed):
+```bash
+/target/release/shielder-cli withdraw --deposit-id 0 --amount 11
+```
+
+A natural first question would be: 'how do I get the depost id'? When your run `shielder-cli show-assets` you also see the deposit ids along the tokens you deposit. In our case, the deposit id will be 0.
+You can now inspect the balance of Damian's account using the Contracts UI to verify that the funds indeed made it to his account.
+
+But here comes a major twist and arguably a large chunk of Shielder's power! We can supply an additional argument: `--recipient`. Please read the next section to find out about what value this adds to the whole setup.
+
+### Transferring tokens
+
+As mentioned previously, when withdrawing assets from the Shielder, we can spefify the additional `--recipient` argument. What it does is withdraw the funds not into the caller's account but into the account specified by the argument, thereby supplying us with extremely useful private transfers.
+Now we can easily (and privately) transfer some tokens to Hans! All that's needed is running the following command:
+```bash
+./target/release/shielder-cli withdraw --deposit-id 0 --recipient 5GBNeWRhZc2jXu7D55rBimKYDk8PGk8itRYFTPfC8RJLKG5o --amount 15
+```
+Note that you will still need to type in the seed of the withdrawer, not the recipient of the transfer (the CLI is quite vocal about this fact, so no need to worry).
+
+You can now check if Hans' balance has been increased accordingly (we bet you 10 TZERO it has!).
+
+### Closing remarks
+
+If you made it this far: congrats! You've just completed your first foray into the land of privacy and zero-knowledge. Now you're ready to shield your tokens and make private transfers!
```

Comment:
hmm technically withdrawing to another account is not the same as private transfers :xd ...

Reply:
Yeah, I was afraid this was going to be inexact, I'll change it to something less controversial :P